java/code/src/com/suse/manager/webui/services/pillar/MinionGeneralPillarGenerator.java

```diff
@@ -56,18 +57,21 @@
      * @return the SaltPillar containing the pillar data
      */
     @Override
-    public Optional<SaltPillar> generatePillarData(MinionServer minion) {
-        SaltPillar pillar = new SaltPillar();
-        pillar.add("org_id", minion.getOrg().getId());
-
-        pillar.add("contact_method", minion.getContactMethod().getLabel());
-        pillar.add("mgr_server", minion.getChannelHost());
+    public Optional<Pillar> generatePillarData(MinionServer minion) {
+        Pillar pillar = minion.getPillars().stream()
+                .filter(item -> item.getCategory().equals(CATEGORY))
+                .findFirst()
+                .orElse(new Pillar(CATEGORY, new HashMap<>(), minion));
+        pillar.getPillar().put("org_id", minion.getOrg().getId());
```

Comment:
As this is repeating pattern, wouldn't it make sense to add `put` or rather `add` to the Pillar itself?

```java
    public Pillar add(String category, Object data) {
        this.pillar.put(category, data);
        return this;
    }
```

So the usage would be the same as before?

Reply:
I pondered doing that... I'll add it then